### ISSUE TITLE
Asks to install esptool every time in WSL2

### ISSUE THREAD

**Issue by mstahl** [abobija/idfx#15]
So, every time I use this to flash my ESP32, it asks me if I want to install esptool. I can't seem to install esptool in enough places to placate idfx, and regardless other than this one thing idfx works perfectly for me. What am I doing wrong over here?

For now I've worked around it by aliasing `idfx build && echo y | idfx flash COM3` but that seems silly.

Thanks for this tool and for helping me out.

**Comment by abobija** [comment]
Hi @mstahl

Thank you for the feedback. It's interesting situation. Can you please execute next commands and give me results so I can investigate what can be the potential cause of the problem in your environment.

Execute next command on Windows to check Python version:

```
python --version
```

Execute next command on Windows to get the list of installed Python modules, and check if there is module `esptool` in the list:

```
pip freeze
```

Execute next command inside of WSL to check does it gets correct information about `esptool` installation status:

```
python.exe -c 'import pkgutil; print(1 if pkgutil.find_loader("esptool") else 0)'
```

**Comment by mstahl** [comment]
Thanks for your reply. From cmd.exe `python --version` outputs `Python 3.9.7`.

Output of `pip freeze` follows: ```
bitstring==3.1.7
cffi==1.14.5
cryptography==3.4.7
ecdsa==0.17.0
esptool==3.1
pycparser==2.20
pyserial==3.5
reedsolo==1.5.4
six==1.16.0```

Here's where it gets a little weird. `python.exe` doesn't have any output when I call it from within WSL. `which python.exe` finds it over in the Windows drive. I dunno if there's an equivalent in CMD to figure out which python.exe is working from there. When I run your command with `python` in WSL instead, I get the expected `1` output.

When I run `pip freeze` in my working directory in WSL, I didn't see esptool in there, installed it, same result.

Like I said it's not a huge deal, since it builds and flashes just fine regardless, it's just a minor annoyance.

**Comment by abobija** [comment]
> Here's where it gets a little weird. `python.exe` doesn't have any output when I call it from within WSL

Yeah, that could be the problem because `idfx` uses `python.exe` internally.

When I execute `python.exe --version` inside of WSL I'm getting correct version of python installed on Windows:

![Screenshot_1](https://user-images.githubusercontent.com/45392201/132007302-66967ae1-06e2-4847-ac80-fee5e7b19184.png)

And when I execute `python --version` in command prompt on Windows, I'm getting the same result, which is OK:

![Screenshot_2](https://user-images.githubusercontent.com/45392201/132007480-16a449c3-08dc-4a85-a63f-b4e79ca559af.png)

You can check did you have Python on the PATH on the Windows, but I assume that you have it because you are getting the path when you execute `which python.exe` in WSL.

Hmm, currently I'm out of ideas...

**Comment by mstahl** [comment]
Ok! I got it. It was, as I had thought, pretty weird. I didn't know about the `where` command in Windows, which helped a lot in debugging (thanks for that!).

What happened is I had a phantom install of Python that looked like it was from the Microsoft Store, but Microsoft Store had somehow lost track of it, so that uninstalling it in Microsoft Store did nothing, and installing there installed another Python in a slightly different location. I'm thinking maybe Windows moved where those apps go when you install them, but that's my best theory. Because "python.exe" from within WSL was getting the weird install, which of course didn't have `esptool`, instead of the normal one where I'd installed it, installing it from either WSL or Windows didn't help. I deleted that extra Python, re-installed Python from the installer at python.org, and now we're good. It no longer asks me to repeatedly install esptool! Thanks again for your help. I probably have to fix my PATH too, but that's a journey I must take alone.

For anyone else having this issue: run `where python.exe` in Windows and `which python.exe` in WSL, and make sure the one listed in WSL is the **first** one listed in Windows.

Thanks again!
